### PR TITLE
Bugfix/crash import wrong private key

### PR DESCRIPTION
**Pull request by fryziuk**
Add test import valid message info as private key
Add test import valid message info with encrypted data as private key
Bugfixes

### CODE REVIEW

File: tests/foundation/data/include/test_data_key_provider.h [hunk 1]
```diff
@@ -32,7 +32,9 @@
 //
 //  Lead Maintainer: Virgil Security Inc. <support@virgilsecurity.com>
 
-
```
Comment (SergeySeroshtan) [inline tests/foundation/data/include/test_data_key_provider.h]: Restore empty line

File: tests/foundation/data/include/test_data_key_provider.h [hunk 2]
```diff
@@ -32,7 +32,9 @@
...
 #include "vsc_data.h"
 
 extern const vsc_data_t test_key_provider_MESSAGE_SHA512_DIGEST;
+extern const vsc_data_t test_key_provider_INVALID_PRIVATE_KEY_VALID_MESSAGE_INFO_WITH_ENCRYPTED_DATA;
+extern const vsc_data_t test_key_provider_INVALID_PRIVATE_KEY_VALID_MESSAGE_INFO;
+
```
Comment (SergeySeroshtan) [inline tests/foundation/data/include/test_data_key_provider.h]: Remove it

File: tests/foundation/data/src/test_data_key_provider.c [hunk 3]
```diff
@@ -49,3 +49,148 @@ const byte MESSAGE_SHA512_DIGEST[] = {
...
+    0x2b, 0x41, 0x03, 0x59, 0xea, 0xca, 0xa3, 0x1c,
+    0x10, 0xfd, 0x26, 0x2b, 0x3e, 0x66, 0x22, 0x7f,
+    0x03, 0x1e, 0x41, 0xc0, 0x20, 0x60, 0xad, 0x83,
+    0x88, 0xda, 0xcd, 0xd3, 0x99, 0x0d, 0x4d, 0xf0,
+    0xb4, 0xe0, 0x47, 0xf0, 0x34, 0x88, 0xac, 0x58,
+    0xa5, 0x9d, 0x01, 0xd5, 0x3d
+};
+
+const vsc_data_t test_key_provider_INVALID_PRIVATE_KEY_VALID_MESSAGE_INFO_WITH_ENCRYPTED_DATA = {
+    INVALID_PRIVATE_KEY_VALID_ASN1_WITH_EXTRA_DATA, sizeof(INVALID_PRIVATE_KEY_VALID_ASN1_WITH_EXTRA_DATA)
```
Comment (SergeySeroshtan) [inline tests/foundation/data/src/test_data_key_provider.c]: Fix names

File: tests/foundation/data/src/test_data_key_provider.c [hunk 4]
```diff
@@ -49,3 +49,148 @@ const byte MESSAGE_SHA512_DIGEST[] = {
...
+    0xFD, 0xA8, 0xF2, 0xF7, 0x0B, 0xCB, 0x71, 0x92,
+    0x0C, 0x60, 0xD0, 0x8C, 0x22, 0x93, 0x69, 0x87,
+    0xA5, 0xB6, 0x80, 0xCF, 0xF0, 0xAB, 0xA9, 0x25,
+    0x89, 0x0C, 0xCD, 0xD6, 0x43, 0x98, 0x0C, 0x05,
+    0x15, 0xC3, 0x4C, 0x3E, 0x79, 0x9A, 0x8F, 0x57,
+    0x88, 0x6B, 0x7E, 0xD5, 0x09
+};
+
+const vsc_data_t test_key_provider_INVALID_PRIVATE_KEY_VALID_MESSAGE_INFO = {
+    INVALID_PRIVATE_KEY_VALID_ASN1, sizeof(INVALID_PRIVATE_KEY_VALID_ASN1)
```
Comment (SergeySeroshtan) [inline tests/foundation/data/src/test_data_key_provider.c]: Fix names

Comment (fryziuk) [reply tests/foundation/data/src/test_data_key_provider.c]: What's wrong with that?

File: tests/foundation/data/src/test_data_key_provider.c [hunk 5]
```diff
@@ -123,7 +123,7 @@ const byte INVALID_PRIVATE_KEY_VALID_ASN1_WITH_EXTRA_DATA[] = {
 };
 
 const vsc_data_t test_key_provider_INVALID_PRIVATE_KEY_VALID_MESSAGE_INFO_WITH_ENCRYPTED_DATA = {
-    INVALID_PRIVATE_KEY_VALID_ASN1_WITH_EXTRA_DATA, sizeof(INVALID_PRIVATE_KEY_VALID_ASN1_WITH_EXTRA_DATA)
+    INVALID_PRIVATE_KEY_VALID_ASN1_WITH_ENCRYPTED_DATA, sizeof(INVALID_PRIVATE_KEY_VALID_ASN1_WITH_ENCRYPTED_DATA)
```
Comment (SergeySeroshtan) [inline tests/foundation/data/src/test_data_key_provider.c]: Again. Wrong names!

**Pull request merged**